GeosCore/photolysis_mod.F90

```diff
@@ -170,7 +170,23 @@ SUBROUTINE INIT_PHOTOLYSIS( Input_Opt, State_Grid, State_Chm, State_Diag, RC )
     !------------------------------------------------------------------------
     CALL RD_AOD( Input_Opt, State_Chm, RC )
     IF ( RC /= GC_SUCCESS ) THEN
-       ErrMsg = 'Error encountered in FAST-JX routine "RD_AOD"!'
+       ErrMsg = 'Error encountered in routine "RD_AOD"!'
+       CALL GC_Error( ErrMsg, RC, ThisLoc )
+       RETURN
+    ENDIF
+
+    !--------------------------------------------------------------------
```

Comment:
I am trying to remember why we need to call this before exiting if photolysis is turned off. Was it to initialize TREF and OREF? Are these used somewhere other than photolysis? If not, maybe we need to make a change somewhere to else to avoid having to call this when photolysis is off.

Reply:
@lizziel: I think it was to avoid an error.  I can double check.

Reply:
!lizziel: We have to call `RD_AOD` and `RD_PROF_NC` to print out the file names that will be opened, for the dry-run.  
```F90
    !------------------------------------------------------------------------
    ! Read in AOD data even if photolysis disabled
    ! (or just print file name if in dry-run mode)
    !------------------------------------------------------------------------
    CALL RD_AOD( Input_Opt, State_Chm, RC )
    IF ( RC /= GC_SUCCESS ) THEN
       ErrMsg = 'Error encountered in routine "RD_AOD"!'
       CALL GC_Error( ErrMsg, RC, ThisLoc )
       RETURN
    ENDIF

    !--------------------------------------------------------------------
    ! Read in T & O3 climatology to fill e.g. upper layers or if O3 not calc.
    !--------------------------------------------------------------------
    ! NOTE: Cloud-J reads in an ascii file with this data during initialization
    ! and uses it prior to calling Cloud_JX within the Cloud-J standalone. In
    ! GEOS-Chem we read a netcdf file instead and use the data within
    ! subroutine Set_Prof_Fjx if using Fast-JX and Set_Prof_CloudJ if using
    ! Cloud-J. The data is stored in State_Chm%Phot%TREF/%OREF. Cloud-J
    ! globals variables TREF and OREF are only used for Cloud-J standalone.
    CALL RD_PROF_NC( Input_Opt, State_Chm, RC )
    IF ( RC /= GC_SUCCESS ) THEN
       ErrMsg = 'Error encountered in routine "Rd_Prof_Nc"!'
       CALL GC_Error( ErrMsg, RC, ThisLoc )
       RETURN
    ENDIF

    !------------------------------------------------------------------------
    ! Exit without doing any computations if we are doing a dry-run
    !------------------------------------------------------------------------
    IF ( Input_Opt%DryRun ) RETURN
```

In both `RD_AOD` and `RD_PROF_NC`, there are shunts to skip reading data if it is a dry-run.

RD_AOD:
```F90
    ! Loop over the array of filenames
    DO k = 1, State_Chm%Phot%NSPAA

       ! Choose different set of input files for standard (trop+strat chenm)
       ! and tropchem (trop-only chem) simulations
       THISFILE = TRIM( DATA_DIR ) // TRIM( SPECFIL(k) )

       !--------------------------------------------------------------
       ! In dry-run mode, print file path to dryrun log and cycle.
       ! Otherwise, print file path to stdout and continue.
       !--------------------------------------------------------------

       ! Test if the file exists
       INQUIRE( FILE=TRIM( ThisFile ), EXIST=FileExists )

       ! Test if the file exists and define an output string
       IF ( FileExists ) THEN
          FileMsg = 'RD_AOD: Opening'
       ELSE
          FileMsg = 'RD_AOD: REQUIRED FILE NOT FOUND'
       ENDIF

       ! Write to stdout for both regular and dry-run simulations
       IF ( Input_Opt%amIRoot ) THEN
          WRITE( 6, 300 ) TRIM( FileMsg ), TRIM( ThisFile )
300       FORMAT( a, ' ', a )
       ENDIF

       ! For dry-run simulations, cycle to next file.
       ! For regular simulations, throw an error if we can't find the file.
       IF ( Input_Opt%DryRun ) THEN
          CYCLE
       ELSE
          IF ( .not. FileExists ) THEN
             WRITE( ErrMsg, 300 ) TRIM( FileMsg ), TRIM( ThisFile )
             CALL GC_Error( ErrMsg, RC, ThisLoc )
             RETURN
          ENDIF
```
RD_PROF_NC:

```F90
    !=================================================================
    ! In dry-run mode, print file path to dryrun log and exit.
    ! Otherwise, print file path to stdout and continue.
    !=================================================================

    ! Test if the file exists
    INQUIRE( FILE=TRIM( nc_path ), EXIST=FileExists )

    ! Test if the file exists and define an output string
    IF ( FileExists ) THEN
       FileMsg = 'RD_PROF_NC: Opening'
    ELSE
       FileMsg = 'RD_PROF_NC: REQUIRED FILE NOT FOUND'
    ENDIF

    ! Write to stdout for both regular and dry-run simulations
    IF ( Input_Opt%amIRoot ) THEN
       WRITE( 6, 300 ) TRIM( FileMsg ), TRIM( nc_path )
300    FORMAT( a, ' ', a )
    ENDIF

    ! For dry-run simulations, return to calling program.
    ! For regular simulations, throw an error if we can't find the file.
    IF ( Input_Opt%DryRun ) THEN
       RETURN
    ELSE
       IF ( .not. FileExists ) THEN
          WRITE( ErrMsg, 300 ) TRIM( FileMsg ), TRIM( nc_path )
          CALL GC_Error( ErrMsg, RC, ThisLoc )
          RETURN
       ENDIF
    ENDIF
```

Reply:
The ordering here with regards to when the routine exits when photolysis is turned off is different than before. CALC_AOD needs to be called regardless of whether J-values will be computed, but RD_PROF_NC does not need to be called if skipping Fast-JX/Cloud-J. But I can clean that up later. I think the aerosols stuff that always needs to be done should be taken out of photolysis_mod and then it will be more clear.